exporter/views.py

```diff
@@ -120,34 +125,37 @@ def get(self, request, *args, **kwargs):
                 f.write(asset.latest_transcription or "")
 
         # Turn Structure into bagit format
-        bagit.make_bag(export_base_dir, {"Contact-Name": request.user.username})
+        bagit.make_bag(export_base_dir)
 
         # Build .zip file of bagit formatted Campaign Folder
         archive_name = export_base_dir
         shutil.make_archive(archive_name, "zip", export_base_dir)
 
-        # Download zip
-        with open("%s.zip" % export_base_dir, "rb") as zip_file:
-            response = HttpResponse(zip_file, content_type="application/zip")
-
-        export_filename = "%s-%s.zip" % (
-            campaign_slug,
-            datetime.today().isoformat(timespec="minutes"),
-        )
-
-        response["Content-Disposition"] = "attachment; filename=%s" % export_filename
-
-        # Upload zip to S3 bucket
-        s3_bucket = getattr(settings, "S3_BUCKET_NAME", None)
-
-        if s3_bucket:
-            s3 = boto3.resource("s3")
-            s3.Bucket(s3_bucket).upload_file(
-                "%s.zip" % export_base_dir, "exporter/%s" % export_filename
+        try:
+            # Upload zip to S3 bucket
+            s3_bucket = getattr(settings, "EXPORT_S3_BUCKET_NAME", None)
+            export_filename = "%s-%s.zip" % (
+                campaign_slug,
```

Comment:
It looks like we could refactor this not to need the `campaign_slug` variable to be passed in if this ZIP call simply used `os.path.basename(export_base_dir)`

Reply:
But then the zip file would have the extra chars in it that are in the temporarily-created directory, wouldn't it?

Reply:
In my testing (I checked our current data for all the download URLs we have so far) the regex that I just committed works to get all the asset IDs. For whatever reason, if I change the * to a ? it fails the unit tests. I'm not really sure I care enough to figure out exactly why. The other regex I tried (allowing all characters except slash) also failed in the unit test.